tests/unit/oplogPopulator/oplogPopulator.js

```diff
@@ -629,202 +529,113 @@ describe('OplogPopulator', () => {
-                name: 'source-connector',
-                config: {},
-            })
-            .then(() => {
-                assert(getConnectorsStub.calledOnce);
-                assert(updateConfStub.calledOnce);
-                assert(createConnectorStub.notCalled);
-            })
-            .catch(err => assert.deepEqual(err, errors.InternalError));
-        });
-
-        it('should throw error if it fails to create connector', async () => {
-            const getConnectorsStub = sinon.stub(oplogPopulator._connectWrapper, 'getConnectors')
-                .resolves([]);
-            const updateConfStub = sinon.stub(oplogPopulator._connectWrapper, 'updateConnectorConfig')
-                .resolves(null);
-            const createConnectorStub = sinon.stub(oplogPopulator._connectWrapper, 'createConnector')
-                .rejects(errors.InternalError);
-            await oplogPopulator._configureConnector({
-                name: 'source-connector',
-                config: {},
-            })
-            .then(() => {
-                assert(getConnectorsStub.calledOnce);
-                assert(updateConfStub.notCalled);
-                assert(createConnectorStub.calledOnce);
-            })
-            .catch(err => assert.deepEqual(err, errors.InternalError));
-        });
-    });
-
-    describe('setup', () => {
-        it('should setup the OplogPopulator', async () => {
-            const setupMongoStub = sinon.stub(oplogPopulator, '_setupMongoClient')
-                .resolves();
-            const setupMetastore = sinon.stub(oplogPopulator, '_setMetastoreChangeStream')
-                .resolves();
-            const getbucketStub = sinon.stub(oplogPopulator, '_getBackbeatEnabledBuckets')
-                .resolves([]);
-            const getDefaultConfigStub = sinon.stub(oplogPopulator, '_getDefaultConnectorConfiguration')
-                .returns(connectorConfig);
-            const generatePipelineStub = sinon.stub(oplogPopulator, '_generateNewConnectorPipeline')
-                .returns('[]');
-            const configureConnectorsStub = sinon.stub(oplogPopulator, '_configureConnector')
-                .resolves();
-            await oplogPopulator.setup()
-            .then(() => {
-                assert(setupMongoStub.calledOnce);
-                assert(getbucketStub.calledOnce);
-                assert(getDefaultConfigStub.calledOnceWithMatch('source-connector'));
-                assert(generatePipelineStub.calledOnceWithMatch([]));
-                assert(configureConnectorsStub.calledOnceWithMatch({
-                    name: 'source-connector',
-                    config: connectorConfig,
-                }));
-                assert(setupMetastore.calledOnce);
-            })
-            .catch(err => assert.ifError(err));
+    describe('_handleChangeStreamChangeEvent', () => {
+        it('should stop listening to bucket if got deleted', done => {
```

Comment:
This series of tests may not require a `done` callback since they look to be synchronous.